code/modules/reagents/reagent_containers/food/snacks/grown.dm

```diff
@@ -1078,3 +1079,93 @@ var/list/special_fruits = list()
 	filling_color = "#DFE88B"
 	plantname = "mustardplant"
 	fragrance = INCENSE_MUSTARDPLANT
+
+//Clovers
+/obj/item/weapon/reagent_containers/food/snacks/grown/clover
+	filling_color = "#247E0A"
+	luckiness_validity = LUCKINESS_WHEN_GENERAL_RECURSIVE
+	var/leaves = 3
+	plantname = "clover"
+
+/obj/item/weapon/reagent_containers/food/snacks/grown/clover/zeroleaf
+	leaves = 0
+
+/obj/item/weapon/reagent_containers/food/snacks/grown/clover/oneleaf
+	leaves = 1
+
+/obj/item/weapon/reagent_containers/food/snacks/grown/clover/twoleaf
+	leaves = 2
+
+/obj/item/weapon/reagent_containers/food/snacks/grown/clover/threeleaf
+	leaves = 3
+
+/obj/item/weapon/reagent_containers/food/snacks/grown/clover/fourleaf
+	leaves = 4
+
+/obj/item/weapon/reagent_containers/food/snacks/grown/clover/fiveleaf
+	leaves = 5
+
+/obj/item/weapon/reagent_containers/food/snacks/grown/clover/sixleaf
+	leaves = 6
+
+/obj/item/weapon/reagent_containers/food/snacks/grown/clover/sevenleaf
+	leaves = 7
+
+
+/obj/item/weapon/reagent_containers/food/snacks/grown/clover/New(atom/loc, custom_plantname)
+	. = ..()
+	update_leaves()
+
+/obj/item/weapon/reagent_containers/food/snacks/grown/clover/proc/update_leaves()
+	switch(leaves)
+		if(3)
+			name = "clover"
+			desc = "A cheerful little herb with three leaves."
```

Comment:
I'd consider refactoring these magic strings into their own enum types. Then you can map the enum types to values instead of having to switch on them

Reply:
This language doesn't have enums, but a similar effect could be achieved with lists. But I would say that the current method is clearer, even if it isn't especially pretty. All three values for a given number of leaves are defined in the same place.